.github/workflows/distribution.yml

```diff
@@ -0,0 +1,195 @@
+name: Make Distribution
+
+on:
+  push:
+    tags:
+      - 'v*.*.*'
+
+
+permissions:
+  contents: read
+
+env:
+  PROJ_VERSION: "9.7.0"
+
+jobs:
+  mkgdaldist:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      attestations: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+      - name: Setup environment
+        shell: bash -l {0}
+        env:
+          DEBIAN_FRONTEND: noninteractive
+        run: |
+            sudo apt update
+            sudo apt-get install -y g++ cmake doxygen enchant-2 python3 python3-dev python3-pip python3-venv libproj-dev swig libsqlite3-dev
+            python3 -m venv create doc_env
+            . doc_env/bin/activate
+            python3 -m pip install -r doc/requirements.txt
+            python3 -m pip install setuptools
+            python3 -m pip install pytest
+            echo PATH=$PATH >> $GITHUB_ENV
+
+      - name: Configure GPG Key
+        run: |
+          echo -n "$GPG_SIGNING_KEY" | base64 --decode | gpg --import
+        env:
+          GPG_SIGNING_KEY: ${{ secrets.GPG_SIGNING_KEY }}
+      - name: Install Doc Dependencies
+        shell: bash
+        run: |
+          sudo apt-get install -y dvipng latexmk texlive-latex-base \
+          texlive-latex-extra git latex-cjk-all texlive-lang-all tex-gyre
+
+      - name: Run mkgdaldist
+        shell: bash -l {0}
+        id: mkgdaldist
+        run: |
+          export RC=$(python3 -c "print('' if 'RC' not in '$github.ref_name' else ''.join('$github.ref_name'.rpartition('RC')[1:].lower()))")
+          export VERSION=$(python3 -c "print('$github.ref_name'.lstrip('v').partition('RC')[0])")
+          export COMPRESSED_VERSION=$(echo "$VERSION" | tr -d .)
+
+          echo "VERSION=$VERSION" >> $env.GITHUB_OUTPUT
+          echo "RC=$RC" >> $env.GITHUB_OUTPUT
+          echo "COMPRESSED_VERSION=$COMPRESSED_VERSION" >> $env.GITHUB_OUTPUT
+
+          if test "$RC" != ""; then
+            ./mkgdaldist.sh $VERSION -tag ${{ github.ref_name }} -rc $RC
+          else
+            ./mkgdaldist.sh $VERSION
+          fi
+      - name: Upload tar.gz
```

Comment:
we'd want to upload the documentation and autotest tar balls/zip as well

Reply:
The artifact upload serves two purposes, make it easier for maintainers to grab these artifacts in case they need to inspect them, and have github compute the sha256 hash as well, which we use in the following steps for `actions/attest-build-provenance`. 

The artifacts that users would generally see are uploaded in the `softprops/action-gh-release@v2` job further down.